### ISSUE TITLE
left join on nil value fails - regression

### ISSUE THREAD

**Issue by genmeblog** [techascent/tech.ml.dataset#377]
```
(let [left (ds/->dataset {:a [nil 2]
                          :b [3 4]})
      right (ds/->dataset {:a [nil 4]
                           :b [6 7]})]
  (j/left-join :a left right))

;; => left-outer-join [2 4]:
;;    | :a | :b | :right.a | :right.b |
;;    |---:|---:|---------:|---------:|
;;    |    |  3 |          |          |
;;    |  2 |  4 |          |          |
```

while the expected result is (6.103):

```
;; => left-outer-join [2 4]:
;;    | :a | :b | :right.a | :right.b |
;;    |---:|---:|---------:|---------:|
;;    |    |  3 |          |        6 |
;;    |  2 |  4 |          |          |
```

**Comment by genmeblog** [comment]
However it works when `:a` is not numerical:

```
(let [left (ds/->dataset {:a [nil "2"]
                          :b [3 4]})
      right (ds/->dataset {:a [nil 4]
                           :b [6 7]})]
  (j/left-join :a left right))
;; => left-outer-join [2 4]:
;;    | :a | :b | :right.a | :right.b |
;;    |----|---:|---------:|---------:|
;;    |    |  3 |          |        6 |
;;    |  2 |  4 |          |          |
```

or:

```
(let [left (ds/->dataset {:a [nil 2]
                          :b [3 4]})
      right (ds/->dataset {:a [nil "4"]
                           :b [6 7]})]
  (j/left-join :a left right))
;; => left-outer-join [2 4]:
;;    | :a | :b | :right.a | :right.b |
;;    |----|---:|---------:|---------:|
;;    |    |  3 |          |        6 |
;;    |  2 |  4 |          |          |
```

**Comment by cnuernber** [comment]
Fixed in 7.012

**Comment by genmeblog** [comment]
lovely! thanks!
